docs/src/references/agent/operation-workflow.md

```diff
@@ -286,4 +288,163 @@ next = ["restarting", "successful_restart", "failed_restart"]
 
 :::note
 This file format is not finalized and will likely be revised.
-:::
\ No newline at end of file
+:::
+
+## Proposal for improvements
+
+### Next step determined by script exit status
+
+The exit status of the script processing a command state
+can be used to determine the next state of the workflow.
+
+The workflow can specify for each exit code:
+- the next command status
+- a failure reason
+
+```toml
+script = "/some/script.sh with some args"
+on_exit.0 = "next_state"                                  # next state for an exit status
+on_exit.1 = { status = "retry_state", reason = "busy"}    # next status with fields 
+on_exit.2-5 = { status = "fatal_state", reason = "oops"}  # next state for a range of exit status
+on_exit._ = "failed"                                      # wildcard for any other non successfull exit
```

Comment:
```suggestion
[current_state.on_exit]
0 = "next_state"                                  # next state for an exit status
1 = { status = "retry_state", reason = "busy"}    # next status with fields 
2-5 = { status = "fatal_state", reason = "oops"}  # next state for a range of exit status
_ = "failed"                                      # wildcard for any other non successfull exit
```
Minor styling suggestion to move the exits codes into a sub-table.

Reply:
It's a matter of preference. Shorter, sure, but also less visually related to the `script` which exit code is observed.

Reply:
Yeah I think keeping the longer version would be better so that it is visually grouped under the state...though of course toml people can do what ever simplification they want